src/Simulation/Microsoft.Quantum.TargetGeneration/GenerateTarget.fs

```diff
@@ -0,0 +1,398 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.QsCompiler.TargetGeneration
+
+open Microsoft.CodeAnalysis.CSharp.Syntax
+
+open Microsoft.Quantum.RoslynWrapper
+
+open System.Collections.Generic
+open System.IO
+open Microsoft.Quantum.QsCompiler.CsharpGeneration
+open Microsoft.Quantum.QsCompiler.SyntaxTree
+open Microsoft.Quantum.QsCompiler.SyntaxTokens
+open Microsoft.Quantum.QsCompiler
+open Microsoft.Quantum.QsCompiler.SyntaxExtensions
+open Microsoft.Quantum.QsCompiler.DataTypes
+open System.Collections.Immutable
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.CSharp
+
+module GenerateTarget =
+    let private autoNamespaces =
+        [
+            "System"
+            "Microsoft.Quantum.Core"
+            "Microsoft.Quantum.Intrinsic"
+            "Microsoft.Quantum.Simulation.Core"
+        ]
+
+    let private intType = ``ident`` "long"
+    let private qubitType = ``ident`` "Qubit"
+    let private qubitArrayType = ``generic`` "IQArray" ``<<`` ["Qubit"] ``>>``
+
+    /// The comment that is displayed at the top of generated files.
+    let private autogenComment = [
+        "//------------------------------------------------------------------------------"
+        "// <auto-generated>                                                             "
+        "//     This code was generated by a tool.                                       "
+        "//     Changes to this file may cause incorrect behavior and will be lost if    "
+        "//     the code is regenerated.                                                 "
+        "// </auto-generated>                                                            "
+        "//------------------------------------------------------------------------------"
+    ]
+
+    // Builds the top-level compilation unit structure
+    let buildHeader (baseClassOpt : string option) ns =
+        let namespaces = match baseClassOpt  with
+                         | Some s -> s.Substring(0, s.LastIndexOf('.')) :: autoNamespaces
+                         | None -> autoNamespaces
+        let usings = namespaces |> List.map (fun ns -> ``using`` ns)
+
+        ``compilation unit``
+            []
+            usings
+            [ns]
+        // We add a "pragma warning disable 1591" since we don't generate doc comments in our C# code.
+        |> ``pragmaDisableWarning`` 1591
+        |> ``pragmaDisableWarning`` 0162 // unreachable code
+        |> ``pragmaDisableWarning`` 0436 // shadowing existing classes from references
```

Comment:
I don't know; again, this is pulled from the C# code gen.

Reply:
Hm, for the C# generation this is there because we re-generate the relevant library code after monomorphization, which will then shadow the code in the referenced library. I don't think this should be applicable in this case, unless I forget something.